pwiz_tools/Skyline/TestFunctional/PastePeptidesWithChargesTest.cs

```diff
@@ -0,0 +1,105 @@
+﻿/*
+ * Original author: Nicholas Shulman <nicksh .at. u.washington.edu>,
+ *                  MacCoss Lab, Department of Genome Sciences, UW
+ *
+ * Copyright 2021 University of Washington - Seattle, WA
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+using System;
+using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using pwiz.Skyline.EditUI;
+using pwiz.Skyline.Model;
+using pwiz.Skyline.Util.Extensions;
+using pwiz.SkylineTestUtil;
+
+namespace pwiz.SkylineTestFunctional
+{
+    /// <summary>
+    /// Verifies that charge indicators (e.g. "+++") can be appended to peptide sequences
+    /// in the Edit > Insert > Peptides dialog.
+    /// </summary>
+    [TestClass]
+    public class PastePeptidesWithChargesTest : AbstractFunctionalTest
+    {
+        [TestMethod]
+        public void TestPastePeptidesWithCharges()
+        {
+            RunFunctionalTest();
+        }
+
+        protected override void DoTest()
+        {
+            // List of peptide sequences, charges, and protein names to paste in the Paste Peptides grid
+            var peptideChargeProteins = new[]
+            {
+                Tuple.Create("GVDC[+57]QEVSQEK", 2, "sp|Q61687|ATRX_MOUSE"),
```

Comment:
Too much repetition of cases. You should add at least one case that will produce more complicated charge indicators.

PEPTIDER, +4
PEPTIDER+++++

Both of these should work, but they are not covered by your test.

Reply:
Is there any other place in Skyline where the user can type in "PEPTIDER, +4" and have it work?
There are places where Skyline displays charges that way, but I cannot find any code in Skyline that knows how to parse that.